cmake/thirdparty/SetupCUDA.cmake

```diff
@@ -74,12 +74,16 @@ message(STATUS "CUDA Libraries:    ${CUDA_LIBRARIES}")
 
 # don't propagate host flags - too easy to break stuff!
 set (CUDA_PROPAGATE_HOST_FLAGS Off)
-if (CMAKE_CXX_COMPILER)
+if (( NOT CUDA_HOST_COMPILER ) AND CMAKE_CXX_COMPILER)
   set (CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
 else ()
   set (CUDA_HOST_COMPILER ${CMAKE_C_COMPILER})
 endif ()
 
+if (CUDA_HOST_COMPILER)
+      list(APPEND CMAKE_CUDA_FLAGS "-ccbin ${CUDA_HOST_COMPILER}")
```

Comment:
Does CMake do this automatically if you set `CMAKE_CUDA_HOST_COMPILER`? https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_HOST_COMPILER.html

Reply:
I tried setting '''CMAKE_CUDA_HOST_COMPILER''' in my hostconfig and it didn't work. Upon further inspection, it seems that it gets wiped out at some point. I'll try to find out where.

Reply:
@davidbeckingsale I didn't make ```CMAKE_CUDA_HOST_COMPILER``` a cache variable in my previous test. If it is set as a cache variable, it works correctly. So why are there ```CMAKE_CUDA_HOST_COMPILER``` and ```CUDA_HOST_COMPILER```?

Reply:
@rrsettgast, the `CUDA_*` flags are used by the `FindCUDA` module, which is now superseded by native support for the CUDA language in CMake. My guess is that this code predates that, hence why `CUDA_HOST_COMPILER` is used here.

We should probably bump the required CMake version for BLT and use the `CMAKE_CUDA_*` flags consistently.

Glad to hear that setting the `CMAKE_CUDA_HOST_COMPILER` flag fixes your issue.

Reply:
Just wanted to add that we should only bump the required CMAKE version to 3.9 or 3.10 for CUDA features when ``ENABLE_CUDA`` is set.

When not using CUDA, I think BLT only currently requires something like 3.3 or 3.5.

Reply:
It is going to be confusing to have different "supported" versions of CMake depending on what feature is enabled etc. I also don't recommend we sprinkle CMake version guards everywhere in BLT -- that is error prone and can get difficult to maintain in the long run.

I suggest the docs are updated to mention that the "recommended" version of CMake is 3.9+, perhaps with a warning (`AUTHOR_WARNING`)